symplyphysics/definitions/linear_coefficient_of_thermal_expansion.py

```diff
@@ -0,0 +1,63 @@
+from sympy import Eq, Derivative
+from symplyphysics import (
+    units,
+    Quantity,
+    Symbol,
+    Function,
+    print_expression,
+    validate_input,
+    validate_output,
+    symbols,
+)
+from symplyphysics.core.geometry.line import two_point_function, Point2D
+
+# Description
+## The coefficient of thermal expansion describes how the size of an object changes with a change in temperature
+## at constant pressure.
+
+# Law: alpha_L = (dL(T)/dT)_p / L(T)
+## alpha_L - linear coefficient of thermal expansion
+## L - body length
+## T - temperature
+## (d/dT)_p - temperature derivative with pressure held constant
+
+# Conditions
+## - Pressure must be constant during the expansion process
+
+linear_expansion_coefficient = Symbol("linear_expansion_coefficient", 1 / units.temperature)
+length = Function("length", units.length)
+temperature = symbols.thermodynamics.temperature
+
+definition = Eq(
+    linear_expansion_coefficient,
+    Derivative(length(temperature), temperature) / length(temperature),
```

Comment:
I am quite sure that `temperature` as derivative parameter, and `temperature` in `length(temperature)` are different temperatures.
For example, most sources use `L0` instead of `length(temperature)`, which is basically `length(temperature_before)`

Reply:
We have similar laws, with the same issue.

Reply:
Not sure if I follow you. Both the derivative $`\frac{d L}{d T}`$ and the original function $`L(T)`$ are calculated at the same temperature point, in theory. The practical (engineering or laboratory) approximation of the derivative might differ in their choice of points and such, but this is a formal mathematical definition.

Reply:
If you use L(t) like described, you will get expansion_function(t). I'm not sure it is really useful this way. Most people expect coefficient at some temperature point, ie at L(t0).
But we cannot take derivative from L(t0), so L(t) in derivative and L(t) after it are two different entities.

Reply:
If this law were about finite changes in temperature and size, i.e. being something like `(length_change / length) / temperature_change`, then yes, we would have to specify the exact temperature points we are calculating these values at. But here the changes are infinitesimal, the functions are continuous, and we can and do actually calculate the value of the derivative of size at the same temperature as the size function. It's basically the rate of change of size divided by the size, both at the same temperature. For engineering or practical purposes, this "true" derivative will be replaced with a discrete one, which is the finite case I wrote above.

Reply:
I understand that the notation $`\frac{d L}{d T}`$ is confusing, but this is basically the same as $`\left.\frac{d L(T')}{d T'}\right|_{T' = T}`$. I've tried this in `sympy` just now and it too makes no difference between the two notations.

Reply:
I'll think about it. I tend to agree with you.
Maybe calling the resulting value a `function` would resolve my confusion.

Reply:
We already defined `length` to be a `Function` and also always write `length(temperature)`. To me it's already a good indication that it's a function.

Reply:
I meant `linear_expansion_coefficient`

Reply:
I meant its name :)

Reply:
I think that would lead to confusion. It's clear it's not a constant, is it

Reply:
Not really. Coefficient looks like some constant value.

Reply:
I suggest writing it in the description. Also, for it to be a constant value, the length function must be an exponential function, which is rarely the case.

Reply:
For constant value it should be taken at some given point t0. That's why I thought t is different under and out of derivative.

Reply:
Why does having $`L(t_0)`$ in the denominator make this coefficient constant? The derivative $`\frac{dL}{dT}`$ is still a function of temperature in general. It _is_ outside the derivative, but they both depend on the same temperature "symbol".

Also, we seem to cute different sources. As far as I can see, Russian ones (I looked in Sivuhin's books) use $`L(0 \degree \mathrm{C})`$ while English ones (say [Springer](https://link.springer.com/chapter/10.1007/978-3-662-02954-1_22)).